simulate/simulate.go

```diff
@@ -335,22 +336,36 @@ func BasesToBaseExt(seq []dna.Base) []BaseExt {
 
 //BaseExtToBases converts a slice of BaseExt to a slice of dna.Base
 func BaseExtToBases(seq []BaseExt) []dna.Base {
-	var newSequence []dna.Base
-	var i, j int
+	var newSequence []dna.Base = make([]dna.Base, len(seq))
 
-	for i = 0; i < len(seq); i++ { //counter to put things back in order
-		for j = 0; j < len(seq); j++ { //counter to check all seq bases before incrementing to look for the next base in the newSeq
-			if seq[j].SeqPos == i {
-				newSequence = append(newSequence, seq[j].Base)
-			}
-		}
+	SortBaseExtBySeqPos(seq)
```

Comment:
No problem here, but for my own understanding why is a sort necessary?

Reply:
this was the only part of simulate that wasn't essentially instantaneous, sort is to put the bases back in order before appending and returning the mutating sequence.